File: apply.js

const fs = require('fs');
const readline = require('readline');

async function processLineByLine() {
  const fileStream = fs.createReadStream('input.txt');

  const rl = readline.createInterface({
    input: fileStream,
    crlfDelay: Infinity
  });

  let regex = /(?<first>\d)(?<second>\d+)/g;
  for await (const line of rl) {
  
    console.log(`Line: ${line}`);
    let result = regex.exec(line);
    if ( ! result ){
        console.log("No match");
    } else {
        do {
            for (let i = 0; i < result.length; i++) {
              console.log(`group ${result[i]}`)
            }
            console.log(`first: ${result.groups.first} second: ${result.groups.second}`)
      
        } while (result = regex.exec(line))
    }

  }
}

processLineByLine();

File: apply.py

import re

regex = re.compile(r"(?P<head>\d)(?P<tail>\d+)")

filename = "input.txt"
with open(filename) as file:
    for line in file:
        print("line: ", line.rstrip())
        for match in re.finditer(regex, line):
            print("match:", match)
            print("groups:", match.groups() )
            print("head:", match.group('head') )
            print("tail:", match.group('tail') )
            

file.close()

